### ISSUE TITLE
patch timeseries group not working

### ISSUE THREAD

**Issue by Enovotny** [USACE/cwms-data-api#879]
I am trying to run the patch timeseries group to add timeseries to a timeseries group.  I am not sure If I am running this correctly. can't seem to get the json correct.

curl -X 'PATCH' \
  'https://cwms-data-test.cwbi.us/cwms-data/timeseries/group/SHEF%20Data%20Acquisition?replace-assigned-ts=false&office=CWMS' \
  -H 'accept: */*' \
  -H 'Authorization: apikey XXX' \
  -H 'Content-Type: application/json' \
  -d '{
  "office-id": "CWMS",
  "id": "SHEF Data Acquisition",
  "time-series-category": {
    "office-id": "CWMS",
    "id": "Data Acquisition",
  },
  "assigned-time-series": [
    {
      "officeId": "LRL",
      "timeseriesId": "Emsworth-EMSP1.Stage-Headwater.Inst.0.0.LPMS-test",
      "aliasId": "OH01.HP.RZZ.0:Units=kcfs"
    }
  ]
}'

I have tried the below as well as which is what the get function provides.

    "assigned-time-series": [      
      {
        "office-id": "LRL",
        "timeseries-id": "Emsworth-EMSP1.Stage-Headwater.Inst.0.0.LPMS-test",
        "alias-id": "OH01.HP.RZZ.0:Units=kcfs"
      }
]

Both give the following errors.

{
  "message": "Formatting error:Could not deserialize: {\n  \"office-id\": \"CWMS\",\n  \"id\": \"SHEF Data Acquisition\",\n  \"time-series-category\": {\n    \"office-id\": \"CWMS\",\n    \"id\": \"Data Acquisition\",\n  },\n  \"assigned-time-series\": [\n      {\n        \"office-id\": \"LRL\",\n        \"timeseries-id\": \"Emsworth-EMSP1.Stage-Headwater.Inst.0.0.LPMS-test\",\n        \"alias-id\": \"OH01.HP.RZZ.0:Units=ft\"\n      },\n      {\n        \"office-id\": \"LRL\",\n        \"timeseries-id\": \"Emsworth-EMSP1.Stage-Tailwater.Inst.0.0.LPMS-test\",\n        \"alias-id\": \"OH01.HT.RZZ.0:Units=ft\"\n      }\n  ]\n} of type: class cwms.cda.data.dto.TimeSeriesGroup",
  "incidentIdentifier": "-5523975055949896025",
  "details": {}
}

**Comment by krowvin** [comment]
Try removing the comma at the end of 

`"id": "Data Acquisition",`

For

`"id": "Data Acquisition"`


Unlike python, JSON standard (parser) is picky with trailing commas. 

VSCode isn't the best at telling you where errors are either. 

For the real sticky ones, I use this 
https://jsonformatter.org/json-parser

**Comment by Enovotny** [comment]
thank you Charles. I can't believe I missed that.  I fixed that and I got back to my original error.  So when running the code below the first time I got a user not authorized.  I had to give myself write permissions to CWMS user in order to get past that error.  I don't think that is correct it should be checking the permission on the assigned-time-series office-id.  At least that is they way it works in CWMS-Vue. Users don't need write permission to CWMS but to the time series office-ID.  

Once I gave myself permission to CWMS I got a 

cwms.cda.api.errors.NotFoundException: ORA-20025: LOCATION_ID_NOT_FOUND: The Location: "Westhope" does not exist.

But Westhope does exist for MVP.  I wonder if it is checking CWMS instead of MVP for this.  By the way I get the same for the code above using an LRL timeseries.  I used the code below as a second check.  

curl -X 'PATCH' \
  'https://cwms-data-test.cwbi.us/cwms-data/timeseries/group/SHEF%20Data%20Acquisition?replace-assigned-ts=false&office=CWMS' \
  -H 'accept: */*' \
  -H 'Authorization: apikey ' \
  -H 'Content-Type: application/json' \
  -d '{
"office-id": "CWMS",
"id": "SHEF Data Acquisition",
"time-series-category": {
"office-id": "CWMS",
"id": "Data Acquisition"
},
"assigned-time-series": [
{
    "office-id": "MVP",
    "timeseries-id": "Westhope.Stage.Inst.15Minutes.0.CEMVP-GOES-Raw",
    "alias-id": "DE32E520.HG.RZZ.0015:TZ=UTC;DLTime=false;Units=ft"
}
]
}'


Also the request body in swagger needs to be updated it shows

  "assigned-time-series": [
    {
      "officeId": "string",
      "timeseriesId": "string",
      "tsCode": 0,
      "aliasId": "string",
      "refTsId": "string",
      "attribute": 0
    }

It should be

  "assigned-time-series": [
    {
      "office-id": "string",
      "timeseries-id": "string",
      "ts-code": 0,
      "alias-id": "string",
      "refTsId": "string",  -> not sure what this should be.
      "attribute": 0
    }
